base_layer/p2p/src/dht_service/dht_service.rs

```diff
@@ -88,52 +88,89 @@ impl DHTService {
     }
 
     /// Construct a new join message that contains the current nodes identity and net_addresses
-    fn construct_join_msg(&self) -> JoinMessage {
-        JoinMessage {
-            node_id: self.node_id.clone(),
-            public_key: self.public_key.clone(),
-            net_address: vec![self.control_service_address.clone()],
-        }
+    fn construct_join_msg(&self) -> Result<JoinMessage, DHTError> {
+        let node_identity = self.node_identity.clone().ok_or(DHTError::NodeIdentityUndefined)?;
```

Comment:
Shouldn't need to clone the Arc here? Can borrow the contents of the option by using `as_ref`

```rust
let node_identity = self.node_identity.as_ref().ok_or(DHTError::NodeIdentityUndefined)?;
```

Reply:
Change applied